### DISCUSSION TITLE
How to get listing for new data objects ?

### DISCUSSION THREAD

**Discussion by nehagaur89** [pimcore/pimcore#14809] [General]
$mapping = new Pimcore\Model\DataObject\CSMapping\Listing();
 
 This is written in my new bundle but I am not sure for vendor folder how to get this path updated to be used in bundle ? I am new to pimcore.. please help. Thanks in advance.

**Comment by podarcis** [comment]
As Pimcore Data Objects classes reside in the `\Pimcore\Model\DataObject` namespace, your invocation is correct. See [Working with Objects via PHP API](https://pimcore.com/docs/pimcore/current/Development_Documentation/Objects/Working_with_PHP_API.html#page_Object-Listings) for examples.

Regarding inheritance you can also optionally [override the models](https://pimcore.com/docs/pimcore/current/Development_Documentation/Extending_Pimcore/Overriding_Models.html) or set an alternative [model parent class](https://pimcore.com/docs/pimcore/current/Development_Documentation/Objects/Object_Classes/Class_Settings/Inheritance.html#page_Parent-Class-Class-Inheritance) and there you would usually use the application namespace or bundle namespace.

**Reply by nehagaur89** [reply]
Hello @podarcis , Thanks I had checked these articles earlier also , but still I am not sure how to proceed. 

The namespace I have mentioned is in vendor folder and that will be in git ignore always , so how to add new code there. How vendor folders are updated for new data objects created. I want to understand that please. Thank you

**Reply by podarcis** [reply]
Data Object classes (your app folder `var/classes/DataObject`) are **generated** by Pimcore based on class definition files (`var/classes/definition_*.php`). You'll never edit those generated files directly and the class definition files are usually edited in backend editor only.

When you write your own bundle you usually provide the JSON-exported class definition files in a directory like `src/YourBundle/Resources/install/classes` and have an [installer routine](https://pimcore.com/docs/pimcore/current/Development_Documentation/Extending_Pimcore/Bundle_Developers_Guide/Pimcore_Bundles/Installers.html) which setups those classes at install time for the user of your bundle. You may want to have a look at a random [community bundle](https://github.com/dachcom-digital/pimcore-members).

**Reply by nehagaur89** [reply]
Thank you !!!
